tools/m2kcli/commands/analog/analog_out.cpp

```diff
@@ -0,0 +1,278 @@
+	while (index < argc && *argv[index] != '-') {
+		std::string argument(argv[index]);
+		if (argument.find('=') == std::string::npos) {
+			throw std::runtime_error("Expecting the following format: <attribute>=<value>\n");
+		}
+		std::istringstream iss(argument);
+		std::string value;
+		getline(iss, argument, '=');
+		getline(iss, value, '=');
+
+		if (argument == "sampling_frequency") {
+			for (int &channel : channels) {
+				analogOut->setSampleRate(channel, std::stod(value));
+			}
+		} else if (argument == "oversampling_ratio") {
+			for (int &channel : channels) {
+				analogOut->setOversamplingRatio(channel, std::stod(value));
+			}
+		} else {
+			throw std::runtime_error("Invalid attribute: " + argument + '\n');
+		}
+		index++;
+	}
+}
+
+const struct option AnalogOut::options[] = {
+	{"help",        no_argument,       nullptr, 'h'},
+	{"quiet",       no_argument,       nullptr, 'q'},
+	{"calibrate",   no_argument,       nullptr, 'C'},
+	{"generate",    required_argument, nullptr, '9'},
+	{"get-channel", required_argument, nullptr, 'G'},
+	{"set-channel", required_argument, nullptr, 'S'},
+	{nullptr, 0,                       nullptr, 0},
+};
+
+
+const char *const AnalogOut::helpMessage = "Usage:\n"
+					   "m2kcli analog-out [uri]\n"
+					   "                  [-h | --help]\n"
+					   "                  [-q | --quiet]\n"
+					   "                  [-C | --calibrate]\n"
+					   "                  [-9 | --generate channel=<index>,... cyclic=<value> raw=<value> [format=<type>] [file=<path>]]\n"
+					   "                  [-G | --get-channel channel=<index> <attribute> ...]\n"
+					   "                  [-S | --set-channel channel=<index> <attribute>=<value> ...]\n"
+					   "\n"
+					   "Pinout:\n"
+					   "\u25A1 \u25A1 \u25A1 \u25A1 \u25A4 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1\n"
+					   "\u25A1 \u25A1 \u25A1 \u25A1 \u25A5 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1\n"
+					   "Positional arguments:\n"
+					   "  uri                   describe the context location \n"
+					   "                        auto | ip:192.168.2.1 | usb:XX.XX.X\n"
+					   "Optional arguments:\n"
+					   "  -h, --help            show this help message and exit\n"
+					   "  -q, --quiet           return result only\n"
+					   "  -C, --calibrate       calibrate the DAC"
+					   "  -9, --generate channel=<index>,... cyclic=<value> raw=<value>[format=<type>] [file=<path>]\n"
+					   "                        generate a signal\n"
+					   "                        by default the samples are read from stdin\n"
+					   "                        after writing all samples press CTRL+D(Unix) or CTRL+Z(Windows)\n"
+					   "                        one channel: channel=<index>\n"
+					   "                        many channels: channel=<index>,<index>,<index>...\n"
+					   "                        channel - {0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15}\n"
```

Comment:
Channels for the analog-out should be 0 and 1.

Reply:
The correct channels were added in commit a61547e.